### PR TITLE
feat(reuse): REUSE best practices for licensing/copyright

### PR DESCRIPTION
**Pull request by rohitpandey49**
Signed-off-by: Rohit Pandey <rohit.pandey4900@gmail.com>

## Description

REUSE best practices for licensing/copyright

### Changes

1. Added License File(a file containing the text of a license) in `LICENSE` folder
2. Implemented correct licensing/copyright format

### Current REUSE status

When running reuse lint over this branch, the result is as follows:

```
* Files with copyright information: 2619 / 4360
* Files with license information: 1066 / 4360
```

Before that we were at:

```
* Files with copyright information: 2616 / 4360
* Files with license information: 503 / 4360
```

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2235"></a>

### CODE REVIEW

File: src/copyright/agent/keyword.conf [hunk 1]
```diff
@@ -1,10 +1,7 @@
 # properties file of the fossology KEYWORD agent
-# Copyright Siemens AG 2018, 2022,
-#
-# Copying and distribution of this file, with or without modification,
-# are permitted in any medium without royalty provided the copyright
-# notice and this notice are preserved.  This file is offered as-is,
-# without any warranty.
+# SPDX-FileCopyrightText: © 018, 2022 Siemens AG
```
Comment (GMishx) [inline src/copyright/agent/keyword.conf:2]:
```suggestion
# SPDX-FileCopyrightText: © 2018, 2022 Siemens AG
```

File: src/copyright/ui/HistogramBase.php [hunk 2]
```diff
@@ -1,21 +1,11 @@
...
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, write to the Free Software Foundation, Inc.,
- * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
- ***********************************************************/
+/*
+ SPDX-FileCopyrightText: © 2014-2017 Siemens AG
+ SPDX-FileCopyrightText: © J. Najjar
+
+ SPDX-License-Identifier: GPL-2.0-only
+ Author: J.Najjar
```
Comment (shaheemazmalmmd) [inline src/copyright/ui/HistogramBase.php:7]: Remove author to maintain consistency.

File: src/copyright/agent/copyscan.cc [hunk 3]
```diff
@@ -1,20 +1,9 @@
...
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
- * See the GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software Foundation,
- * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
- */
+ SPDX-FileCopyrightText: © 2015 Siemens AG
+ SPDX-FileCopyrightText: © Florian Krügel
```
Comment (NicolasToussaint) [inline src/copyright/agent/copyscan.cc:3]: beware, here you turned an `author` into a `copyright`, these are not the same.
the Author line should be kept unchanged.

File: src/copyright/agent_tests/Functional/shunit2 [hunk 4]
```diff
@@ -2,14 +2,13 @@
...
-# Copyright 2008 Kate Ward. All Rights Reserved.
-# Released under the LGPL (GNU Lesser General Public License)
+# SPDX-FileCopyrightText: © 2008 Kate Ward <kate.ward@forestent.com>
+#
+# SPDX-License-Identifier: LGPL-2.0-only
 #
 # shUnit2 -- Unit testing framework for Unix shell scripts.
 # http://code.google.com/p/shunit2/
 #
-# Author: kate.ward@forestent.com (Kate Ward)
```
Comment (NicolasToussaint) [inline src/copyright/agent_tests/Functional/shunit2:11]: Author line should not be removed

File: src/copyright/keyword.conf [hunk 5]
```diff
@@ -1,7 +1,4 @@
-; Copying and distribution of this file, with or without modification,
-; are permitted in any medium without royalty provided the copyright
-; notice and this notice are preserved. This file is offered as-is,
-; without any warranty.
+;SPDX-License-Identifier: FSFAP
```
Comment (NicolasToussaint) [inline src/copyright/keyword.conf:1]: same as src/copyright/ecc.conf

File: src/copyright/agent_tests/Functional/shunit2 [hunk 6]
```diff
@@ -2,14 +2,13 @@
 # $Id: shunit2 335 2011-05-01 20:10:33Z kate.ward@forestent.com $
 # vim:et:ft=sh:sts=2:sw=2
 #
-# Copyright 2008 Kate Ward. All Rights Reserved.
-# Released under the LGPL (GNU Lesser General Public License)
+# SPDX-FileCopyrightText: © 2008 Kate Ward <kate.ward@forestent.com>
```
[the remark this answers is not in the document]
Comment (rohitpandey49) [reply src/copyright/agent_tests/Functional/shunit2:5]: Sir, as per this [blog](https://matija.suklje.name/how-and-why-to-properly-write-copyright-statements-in-your-code#what-is-wrong-with-all-rights-reserved), we should not use`All Rights Reserved.` in copyright statement. Please see.

File: src/copyright/agent/copyscan.cc (hunk 3 above)
Comment (rohitpandey49) [reply src/copyright/agent/copyscan.cc]: Corrected.

File: src/copyright/agent/copyscan.hpp [hunk 7]
```diff
@@ -1,21 +1,9 @@
...
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
- * See the GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software Foundation,
- * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
- */
+ SPDX-FileCopyrightText: © 2015 Siemens AG
+ SPDX-FileCopyrightText: © Florian Krügel
```
[the remark this answers is not in the document]
Comment (rohitpandey49) [reply src/copyright/agent/copyscan.hpp]: Corrected.

File: src/copyright/agent_tests/Functional/shunit2 (hunk 6 above)
[the remark this answers is not in the document]
Comment (NicolasToussaint) [reply src/copyright/agent_tests/Functional/shunit2:5]: I meant to say that we should, as part of those commits, transform the statement into REUSE compliant statement, and not change the meaning behind.

But yes, you're right, that `All Rights Reserved` is total nonsense, I agree to remove it here.

[the remark this answers is not in the document]
Comment (rohitpandey49) [reply src/copyright/agent_tests/Functional/shunit2:5]: Sure

**Comment by GMishx** [comment]
@rohitpandey49 can you please squash all 3 commits into 1?

File: src/decider/Makefile [hunk 8]
```diff
@@ -1,9 +1,6 @@
-# Copyright Siemens AG 2014
-#
-# Copying and distribution of this file, with or without modification,
-# are permitted in any medium without royalty provided the copyright
-# notice and this notice are preserved.  This file is offered as-is,
-# without any warranty.
+SPDX-FileCopyrightText: © 2014 Siemens AG
+
+SPDX-License-Identifier: FSFAP
```
Comment (GMishx) [inline src/decider/Makefile:3]:
```suggestion
# SPDX-FileCopyrightText: © 2014 Siemens AG

# SPDX-License-Identifier: FSFAP
```

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by rohitpandey49** [comment]
> @rohitpandey49 can you please squash all 3 commits into 1?
Done

**Pull request merged**